dynamiqs/time_array.py

```diff
@@ -245,6 +241,14 @@ def conj(self) -> TimeArray:
             New time-array object with element-wise complex conjuguated values.
         """
 
+    @abstractmethod
+    def as_batched_callable(self) -> callable[[float], Array]:
+        """Returns the time array under the form of a callable that is vmappable.
```

Comment:
Should the function be called `as_vmappable_callable` instead? Maybe more explicit seeing the one-liner below.